datacompy/core.py

```diff
@@ -297,7 +297,7 @@ def _intersect_compare(self, ignore_spaces, ignore_case):
         """
         LOG.debug("Comparing intersection")
         row_cnt = len(self.intersect_rows)
-        for column in self.intersect_columns():
+        for column in sorted(self.intersect_columns()):
```

Comment:
A sorted column order is deterministic, but I really think you ought to use an ordered set structure here, because doing so will maintain the order of the columns as per the source file. Also, when doing the intersection I imagine there is a way to do a "stable" intersection, akin to a stable sort, which will maintain the column order of the source insofar as the columns match.

Let's say I have a list of people, with the following columns:

|Name|DateOfBirth|Address|City|State|ZipCode|
----------------------------------------------------

And let's say for example's sake the other file being compared against has identical columns. I'd probably want to make the columns keep the same order, for readability, rather than reordering to this:

|Address|City|DateOfBirth|Name|State|ZipCode|
----------------------------------------------------

Reply:
Here is the area that I think should use an ordered set and to the extent possible a "stable" intersection:

https://github.com/capitalone/datacompy/blob/8d96429c9cf574f8d448a4e63301f0025ad467ee/datacompy/core.py#L212-L214

Reply:
I was going for the simplest way to make the output deterministic without adding extra dependencies, as a quick way of obtaining an incremental improvement over the previous implementation. 

I think there are definitely improvements over the simple sorted list I have implemented; please open a PR with your suggested changes!